awkward/array/jagged.py

```diff
@@ -41,21 +40,23 @@ def counts2offsets(cls, counts):
 
     @classmethod
     def offsets2parents(cls, offsets):
-        out = cls.numpy.zeros(offsets[-1], dtype=cls.JaggedArray.fget(None).INDEXTYPE)
-        cls.numpy.add.at(out, offsets[offsets != offsets[-1]][1:], 1)
-        cls.numpy.cumsum(out, out=out)
-        if offsets[0] > 0:
-            out[:offsets[0]] = -1
-        return out
+        dtype = cls.JaggedArray.fget(None).INDEXTYPE
+        counts = cls.numpy.diff(cls.numpy.r_[dtype.type(0), offsets])  # numpy >= 1.16: diff prepend
```

Comment:
Wait—if this is a Numpy 1.16 feature, it can't be the only way to evaluate this function. Awkward depends on `numpy>=1.13.1` and that's an absolute minimum because it's the earliest with `__array_ufunc__`.

(I would like to have gone much earlier—I used to consider it normal to install Python and Numpy from apt-get and everything else from pip. On a brand new Debian box, that means Numpy 1.12.1. For most Numpy work, you can go as early as Numpy 1.4 without noticing much difference in the API. I considered the minimum of 1.13.1 to be a concession, forcing users to get Numpy from pip or conda.)

Somehow, the tests didn't fail for this: our testing matrix covers versions of Python and versions of Numpy, so more than half of them should have failed.

Since the old and new code compute the same function, it wouldn't be unreasonable to branch this by testing for the Numpy feature you need. Is it the _existence_ of `numpy.diff` or `numpy.r_` (both unfamiliar to me)? That's an easy `hasattr` test, and safer than checking version numbers.

Reply:
No, the numpy 1.1.6 feature is `np.diff(array, prepend=0)`.  Because of that, I used an alternative solution that is compatible back to 1.13.  `np.r_[0, a]` is equivalent (but apparently slightly faster than) `np.concatente((0, a))`.

Reply:
Oh, so that's why it worked. I was a little worried that `offsets2parents` never happened anywhere in the tests—it was hard to see how it could have slipped through.

Reply:
FYI `startsstops2parents` didn't happen anywhere in the tests before this PR.  Might be worth doing a coverage check.